### PR TITLE
Add "selected" column to metadata TSV downloads

### PR DESCRIPTION
**Pull request by jgadling**
### Description

This adds an additional "selected" column to metadata tsv downloads in order to improve visualization options in auspice. Only samples that are inputs to the phylo run workflow (translation: only selected samples in on-demand tree builds, or all county samples for nightly runs), or listed as gisaid accessions to be include will have a "yes" value in the selected column.

Note that this pr ALSO updates our test infrastructure to use localstack instead of moto in our tests. This has some drawbacks (s3 state can persist between test runs!) but makes our tests more consistent since all other AWS services our tests interact with (secrets manager, ssm, step functions) are calling localstack as well.

#### Issue
[ch155886](https://app.clubhouse.io/genepi/story/155886)

### Test plan

Unit tests are included and passing. I added a CLI command to download metadata files that works in local dev. I'll test this in staging as well once it's merged.

### CODE REVIEW

File: src/backend/aspen/app/views/phylo_trees.py [hunk 1]
```diff
@@ -167,6 +167,34 @@ def sample_filter(sample: Sample):
...
+        .outerjoin(entity_alias, PhyloRun.inputs.of_type(entity_alias))
+        .outerjoin(Sample)
+        .filter(PhyloTree.entity_id == phylo_tree_id)
+        .options(
+            contains_eager(PhyloTree.producing_workflow.of_type(PhyloRun))
+            .contains_eager(PhyloRun.inputs.of_type(entity_alias))
+            .contains_eager(entity_alias.sample)
+        )
+        .one()
+    )
```
Comment (lvreynoso) [inline src/backend/aspen/app/views/phylo_trees.py:187]: Thanks for your hard work figuring out this query! I know it was not easy.

**Review by lvreynoso — APPROVED** [review]
looks great!!

File: src/backend/aspen/app/views/phylo_trees.py (hunk 1 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/phylo_trees.py:187]: ditto!

File: src/backend/aspen/app/views/phylo_trees.py [hunk 2]
```diff
@@ -181,7 +209,7 @@ def phylo_tree(phylo_tree_id: int):
 
 
 def _extract_accessions(accessions_list: list, node: dict):
-    node_attributes = node["node_attrs"]
+    node_attributes = node.get("node_attrs", {})
```
Comment (phoenixAja) [inline src/backend/aspen/app/views/phylo_trees.py:212]: yay no keyerror 😆

File: src/backend/aspen/app/views/tests/test_auspice_redirect.py [hunk 3]
```diff
@@ -100,8 +105,8 @@ def test_auspice_redirect_view(session, app, client, mock_s3_resource, test_data
     res = client.get(f"/api/auspice/view/{phylo_tree.id}")
     res_presigned = res.json["url"]
 
-    # this is a little hacky, currently when calling a get request on moto generated presigned url it gets 404
-    # i think this is due to calling get not being within the moto test scope, so as a workaround i''m checking that
+    # this is a little hacky, currently when calling a get request on localstack generated presigned url it gets 404
+    # i think this is due to calling get not being within the localstack test scope, so as a workaround i''m checking that
```
Comment (phoenixAja) [inline src/backend/aspen/app/views/tests/test_auspice_redirect.py:109]: thanks for all the comments

File: src/backend/aspen/app/views/phylo_trees.py (hunk 1 above)
Comment (phoenixAja) [reply src/backend/aspen/app/views/phylo_trees.py:187]: ditto!

**Pull request merged**